### DISCUSSION TITLE
a clock, a movement rechargeable system, and...

### DISCUSSION THREAD

**Discussion by IHOXOHI** [micropython/micropython#16326] [Ideas]
Hey,
I have the project to make a  'home-made-micropython-digital-clock' which can be rechargeable by a system of movement (from a human if possible, I think).
For moment, a board could use only 2mA to keep the time on a  litlle display,  I tested the super-capacitor which could be litlle and easy to use... It seems possible... I'm looking for a system which can produce this energy... It seem's that a classical automatic watch use a mecanic system to move a metallic part which could be an aimant...? I tested different 'manual' rechargeable system; they can produce enought energy too...
My opinion is that it could be easy, too, to make a system which could work with the body and produce just 30mA for example...
If someone have a model to expose... It's about magnetic relations, which part of the body is the better choice for example... 
I will start with an easy step, a human on a bicycle... 
Thanks.

**Comment by peterhinch** [comment]
A conventional permanent magnet brushed DC motor works as a generator if you spin the shaft...

**Comment by bidrohini** [comment]
Generating Energy from human motion you can consider using  Piezoelectric materials.

**Reply by IHOXOHI** [reply]
Thanks, I will study it...

**Reply by KnorkFox** [reply]
Piezos can make voltage, but not much power. This is the wrong approach for energy generation. You need a moving magnet and a coil, at least.

**Comment by scruss** [comment]
The term you're looking for is [Energy harvesting](https://en.wikipedia.org/wiki/Energy_harvesting) (English) / [récolte d'énergie](https://fr.wikipedia.org/wiki/R%C3%A9colte_d%27%C3%A9nergie) (French). Many of the ultra-low power energy harvesting devices operate in the µA range, which is typically lower than many MicroPython processors need. Maybe one of the [nrf](https://micropython.org/download/?port=nrf) processors might be micro-power capable, but it's going to have to be in deep sleep most of the time.

I'm about a decade out of date with what the available technologies are, but it's a whole subculture in electronics development.

**Reply by IHOXOHI** [reply]
Thanks, it's the purpose: keep the board in sleep mode most of the time. 
For example I use a ucontroller  to open my home-door : it has to work less than 1 minute per day... I have too recharge it only once per month!

**Comment by jonnor** [comment]
Do you need it to charge up from regular wear? If dedicated effort is suitable, then a bicycle dynamo is a nice standard component that can generate quite a lot of energy rather easily. But the most ready-to-run option is probably a hand-cranked flashlight or USB powerbank. They are typically designed to be quite simple and cheap, so even if you are interested in DIY approach - I think they are worth testing out and studying.

**Reply by IHOXOHI** [reply]
Yes, I have been on the bicycle market... Effectively the "pignon dynamo" seems to do perfectly the job... Thanks.

**Comment by 2dof** [comment]
it's all about to get at least resonable fast spin of generator:
in ‘self-winding’ movements  a balance (mass) and  spring is use to storage energy, but for power a generator (like vibrator motor from phone) aditional release (trigger) is needed to get fast rotation of motor. ( adding small portion of energy and then  fast release)

Many years ago my friend  tested pneumatic solutions  form walking (air pump in shoes) - then directly short, high pressure impulse to turbine (generator)  to  use arm/leg joint movement to  build presure and release - all not comfortable and not practical  but he  got some power. ( he ended op with  crank hand generator but instrad of crank he used linear move of cable betwen two wheels )

**Reply by IHOXOHI** [reply]
Nice thanks. There is model very cheap which work with a press; it seems to be a good try...

**Comment by peterhinch** [comment]
At risk of being OT I couldn't resist recalling this. Many years ago a friend took a beach holiday in Greece to unwind from his hectic life. He had a self-winding mechanical wristwatch which had served him well for many years.

On his return we asked how the holiday had gone. He announced that he'd been so indolent that his self-winding watch stopped.

He never entirely lived it down.

**Reply by IHOXOHI** [reply]
Maybe, it's really the bigger 'default' of what I want to do, or no to do...
